### DISCUSSION TITLE
How do I get a range compare and set pixel value?

### DISCUSSION THREAD

**Discussion by oldercoder** [libvips/libvips#2759] [Q&A]
I saw the C++ document that indicate this :

```C++
VImage a = (a < 128).ifthenelse (128, a);  
```

will set every band element of a less than 128 to 128. But how do I achieve a range ifthenelse invocation? For example:

```C++
VImage a = ((a >128)) &&(a < 200)).ifthenelse (fix_value, a);  // this won't work
```

Thanks for any thoughts!

**Comment by jcupitt** [comment]
Hi @oldercoder, try:

```C++
auto a = ((a >128)) & (a < 200)).ifthenelse(fix_value, a);
```

The short-circuit operators (`&&` and `||`) don't overload well.

**Reply by oldercoder** [reply]
Thanks @jcupitt , that works!
